cpp/src/arrow/flight/internal.cc

```diff
@@ -266,6 +281,12 @@ grpc::Status ToGrpcStatus(const Status& arrow_status, grpc::ServerContext* ctx)
       const std::string detail_string = arrow_status.detail()->ToString();
       ctx->AddTrailingMetadata(internal::kGrpcStatusDetailHeader, detail_string);
     }
+    if (arrow_status.detail()) {
+      auto fsd = FlightStatusDetail::UnwrapStatus(arrow_status);
```

Comment:
UnwrapStatus already takes care of when Status has a null StatusDetail: https://github.com/apache/arrow/blob/fd51e9d818036f57bd2da213dfe20c35597f1add/cpp/src/arrow/flight/types.cc#L67-L73

Reply:
fixed